internal/pkg/aws/sessions/sessions.go

```diff
@@ -104,6 +114,12 @@ func (p *Provider) FromProfile(name string) (*session.Session, error) {
 	if aws.StringValue(sess.Config.Region) == "" {
 		return nil, &errMissingRegion{}
 	}
+	if _, credErr := p.sessionValidator.ValidateCredentials(sess); credErr != nil {
+		if strings.Contains(credErr.Error(), "context deadline exceeded") {
```

Comment:
Does this not work?
```suggestion
		if errors.Is(err, context.DeadlineExceeded) {
```

Reply:
Yeah it doesn't work that is why I had to do the string comparison way

Reply:
Also it is little confusing how sometimes we get ```DealineExceed``` error and other times we get 
```
NoCredentialProviders: no valid providers in chain
caused by: EnvAccessKeyNotFound: failed to find credentials in the environment.
SharedCredsLoad: failed to load profile, .
EC2RoleRequestError: no EC2 instance role found
caused by: EC2MetadataError: failed to make EC2Metadata request
```

So we need to check for both the errors.

Reply:
this is why it doesn't work 😔  https://github.com/aws/aws-sdk-go/issues/2820